File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/supplier.py
from __future__ import annotations

from pydantic import BaseModel


class Supplier(BaseModel):
    supplierId: int
    name: str
    description: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None
    active: bool
    verified: bool


class CreateSupplierRequest(BaseModel):
    name: str
    description: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None
    active: bool = True
    verified: bool = False


class UpdateSupplierRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None
    active: bool | None = None
    verified: bool | None = None

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/Supplier.cs
namespace OctoSupply.Api.Models;

public sealed record Supplier(
    int SupplierId,
    string Name,
    string? Description,
    string? ContactPerson,
    string? Email,
    string? Phone,
    bool Active,
    bool Verified);

public sealed class CreateSupplierRequest
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ContactPerson { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public bool Active { get; set; } = true;
    public bool Verified { get; set; } = false;
}

public sealed class UpdateSupplierRequest
{
    public string? Name { get; set; }
    public string? Description { get; set; }
    public string? ContactPerson { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public bool? Active { get; set; }
    public bool? Verified { get; set; }
}
